docs/samples/Microsoft.ML.Samples/Program.cs

```diff
@@ -1,12 +1,43 @@
-﻿using Samples.Dynamic;
+﻿using System;
+using System.Reflection;
+using Samples.Dynamic;
 
 namespace Microsoft.ML.Samples
 {
-    internal static class Program
+    public static class Program
     {
-        static void Main(string[] args)
+        public static void Main(string[] args)
         {
             CalculateFeatureContribution.Example();
+
+            if (args.Length > 0 && args[0] == "-runall")
+                RunAll();
+        }
+
+        internal static void RunAll()
+        {
+            foreach (var type in Assembly.GetExecutingAssembly().GetTypes())
+            {
+                var method = type.GetMethod("Example", BindingFlags.NonPublic | BindingFlags.Public
+                    | BindingFlags.Static | BindingFlags.FlattenHierarchy);
+
+                if (method != null)
+                {
+                    try
+                    {
+                        method.Invoke(null, null);
+                    }
+                    catch (Exception ex)
+                    {
+                        // Print the exception for debugging.
+                        Console.Write(type);
+                        Console.Write(ex);
+
+                        // Throw to fail.
+                        throw ex;
```

Comment:
So, generally, if you have an exception thrown it will print out the stack trace and message anyway, just by virtue of you running it. So, I'm not sure why we have this `try`/`catch` block? The only thing that it seems to accomplish is that it makes it 

Also, even if you wanted to do this, which, I really don't think you want to do, which I'm not sure why you do, please just re-throw the exception, so `throw;` instead of `throw ex;`.

So let's just point out, even if we do things this way:

```csharp
private static void EvilMethod()
{
    string a = null;
    Console.WriteLine(a.Length);
}

private static void Main(string[] args)
{
    try
    {
        EvilMethod();
    }
    catch
    {
        throw;
    }
}
```

If I run the debugger, the debugger stops on that last `throw;`. I have no information on anything...

If on the other hand, I just run this simpler code, under the debugger I get it stopped right where the exception was thrown (so I have some ability to inspect the actual state of the objects). Also from the command line perspective even if run without debugging I still get the console.

```csharp
private static void EvilMethod()
{
    string a = null;
    Console.WriteLine(a.Length);
}

private static void Main(string[] args)
{
    EvilMethod();
}
```

So, it seems like the only thing all this extra material accomplishes is, it just makes it harder to debug the actual issue?

Reply:
I've never seen a valid case of `throw ex;` in a catch block in my time as a developer. It always should have been just `throw;`